front_end/elements/DOMPath.js

```diff
@@ -60,8 +60,12 @@ Elements.DOMPath._cssPathStep = function(node, optimized, isTargetNode) {
   }
   const nodeName = node.nodeNameInCorrectCase();
 
-  if (id)
-    return new Elements.DOMPath.Step(nodeName + idSelector(id), true);
+  if (id) {
+    const test = new Elements.DOMPath.Step(nodeName + idSelector(id), true);
+    if (document.querySelectorAll(test.value).length == 1) {
```

Comment:
this `document` is not page's document: it's devtools frontend itself. You'd rather want to run protocol method instead, but it's getting way more involved.

Reply:
whoops i was using it in extension, thanks for pointing it out, i modified the pull request, now probably it works.

using `node.ownerDocument` instead of `document`